src/decisionengine/framework/taskmanager/TaskManager.py

```diff
@@ -122,38 +128,59 @@ def __init__(self, channel_dict, channel_name):
 
         self.le_s = _make_workers_for(configured_le_s, LogicEngine, channel_name)
 
-        delogger.debug("Creating channel transform")
+        delogger.debug("Creating channel transforms")
         transform_workers = _make_workers_for(channel_dict["transforms"], Transform, channel_name)
         self.transform_workers = ensure_no_circularities(self.source_workers, transform_workers, self.publisher_workers)
         self.task_manager = channel_dict.get("task_manager", {})
 
 
 class TaskManager(ComponentManager):
     """
-    Task Manager
+    Task manager
     """
 
     def __init__(self, name, generation_id, channel_dict, global_config):
         """
         :type name: :obj:`str`
         :arg name: Name of channel corresponding to this task manager
         :type generation_id: :obj:`int`
-        :arg generation_id: Task Manager generation id provided by caller
+        :arg generation_id: Task manager generation id provided by caller
         :type channel_dict: :obj:`dict`
         :arg channel_dict: channel configuration
         :type global_config: :obj:`dict`
         :arg global_config: global configuration
         """
-        if "channel_name" in channel_dict:
-            self.name = channel_dict["channel_name"]
-        else:
-            self.name = name
+        self.name = channel_dict.get("channel_name", name)
 
         super().__init__(self.name, generation_id, global_config)
         self.logger = structlog.getLogger(CHANNELLOGGERNAME)
         self.logger = self.logger.bind(module=__name__.split(".")[-1], channel=self.name)
         self.workflow = Workflow(channel_dict, self.name)
         self.lock = threading.Lock()
+        self.broker_url = global_config.get("broker_url", "memory:///")
+
+        self.exchange = Exchange("example_topic_exchange", "topic")
+        self.connection = Connection(self.broker_url)
+
+        # Caching fun...
+        #   Just keeping track of instance names will not work
+        #   whenever we have multiple source instances of the same
+        #   source type.
+        expected_source_products = set()
+        queues = {}
+        for worker in self.workflow.source_workers.values():
+            expected_source_products.update(worker.module_instance._produces.keys())
+            queues[worker.full_key] = Queue(
```

Comment:
Can I get a logger.debug "making queue such and such with routing key xyz"?

Reply:
Will do.  Is there a reason to provide a routing key if we're going to have queues dedicated to each source?

Reply:
I believe the routing key is required by the interface.